src/System.IO.Pipelines/PipeFactory.cs

```diff
@@ -99,7 +113,11 @@ public IPipeWriter CreateWriter(IPipeWriter writer, Func<IPipeReader, IPipeWrite
         public IPipeReader CreateReader(IPipeReader reader, Func<IPipeReader, IPipeWriter, Task> produce)
         {
             var pipe = new Pipe(_pool);
-            produce(reader, pipe);
+
+            produce(reader, pipe).ContinueWith(t =>
+            {
+            });
```

Comment:
Or follow the pattern in `AsPipelineWriter` that's been removed where `task.IsFaulted` is checked; but try/catch is as per the other `CreateReader` probably easier to reason about